Update mjml to 3.3.2.

qkevinto opened the pull request.

qkevinto committed:
Update mjml to 3.3.2.

qkevinto committed:
Don't use arrow function with the exports function as it does not get scoped to the webpack object, and handle outputting of mjml errors.

qkevinto committed:
Update readme with correct loader reference and template is not a function.

qkevinto committed:
Return html instead of mjml object to match loader's previous behaviour.

ramasilveyra commented on index.js:
```diff
@@ -1,7 +1,20 @@
 const mjml = require('mjml');
+const colors = require('colors');
```
This require isn't used, could you remove it please?

qkevinto replied:
I'm using this package to colourize the outputted error. The documentation for the [color](https://www.npmjs.com/package/colors) package notes that it extends `String.prototype` hence why I could use `.red` method on a string on [Line 10](https://github.com/ramasilveyra/mjml-loader/pull/1/files/980c1381b4ae14b5d28159921be3ebf0da115aaf#diff-168726dbe96b3ce427e7fedce31bb0bcR10) 😆. But I'll make it more explicit by using the package function and using the colour's method instead.

ramasilveyra replied:
It's true, my bad! Explicit it's better

ramasilveyra commented on package.json:
```diff
@@ -22,6 +22,7 @@
   "homepage": "https://github.com/ramasilveyra/mjml-loader#readme",
   "dependencies": {
-    "mjml": "^2.3.0"
+    "colors": "^1.1.2",
```
Also could you remove this dep from the package.json please?

qkevinto replied:
Please see my comment above regarding this package.

ramasilveyra commented on index.js:
```diff
@@ -1,7 +1,20 @@
-  this.cacheable && this.cacheable();
-  this.value = content;
-  return `module.exports = ${JSON.stringify(mjml.mjml2html(content))};`;
+module.exports = function(content) {
```
Good catch! One little comment, if you want could you please add a name to the anonymous function please?

```js
module.exports =  function mjmlLoader(content) {
```

So debugging tools work better with mjml-loader

ramasilveyra commented on index.js:
```diff
@@ -1,7 +1,20 @@
+  const result = mjml.mjml2html(content);
+
+  if (result.errors.length) {
+    console.log(`[mjml-loader] ERROR in ${this.resourcePath}:`.red);
```
Wow I didn't know about the `.red` on the `console.log`!! Really good!

qkevinto replied:
See my other comment above hehe.

ramasilveyra reviewed (commented):
Thanks! I'm eager to merge this, there is a lot of improvements!

I left some comments, also I see that your commits aren't associated to your GitHub account if this is unintentional you can fix this with https://help.github.com/articles/why-are-my-contributions-not-showing-up-on-my-profile/#you-havent-added-your-local-git-commit-email-to-your-profile + recreating the commits or http://stackoverflow.com/a/3042512/4709891.

And if you don't have time to finish this let me know, I can make time to finish it.

Thanks again!

qkevinto committed:
Changes based on code review.

qkevinto commented:
Ahh, thanks for the note on commits not being associated with my account. I was using a different machine. 🙂

I believe I have addressed all your comments, thanks for the review! Let me know if there's anything else.

ramasilveyra commented:
Thanks for your time on this! Released under `1.0.0-beta.1`
